### PR TITLE
[cpuinfo,fbgemm,nnpack] update to latest source version and rename targets

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?

* Update `cpuinfo` to use the latest source code.
    * Use different log level for debug build.
    * Embed a patch that works like `cpuinfo-pull-22-868bd11.patch`
* Change `CONTROL` to `vcpkg.json`.
* Rename `IMPORTED` target namesapce `cpuinfo::` to `unofficial::cpuinfo::`
  * check below comments
  * The change will affect 
    * merged: #16346 #16344 
    * in-progress: #17064 #17065

The previous patch `cpuinfo-pull-22-868bd11.patch`, which is from https://github.com/pytorch/cpuinfo/pull/22 failed to apply.

```
Starting package 1/1: cpuinfo:x64-osx
Building package cpuinfo[core]:x64-osx...
-- Using cached /Users/user/vcpkg/downloads/cpuinfo-pull-22-868bd11.patch
-- Using cached /Users/user/vcpkg/downloads/pytorch-cpuinfo-5916273f79a21551890fd3d56fc5375a78d1598d.tar.gz
-- Extracting source /Users/user/vcpkg/downloads/pytorch-cpuinfo-5916273f79a21551890fd3d56fc5375a78d1598d.tar.gz
-- Applying patch /Users/user/vcpkg/downloads/cpuinfo-pull-22-868bd11.patch
CMake Error at scripts/cmake/z_vcpkg_apply_patches.cmake:57 (message):
  Applying patch failed: Checking patch CMakeLists.txt...

  error: while searching for:

  OPTION(CPUINFO_BUILD_MOCK_TESTS "Build cpuinfo mock tests" ON)

  OPTION(CPUINFO_BUILD_BENCHMARKS "Build cpuinfo micro-benchmarks" ON)

  

  # ---[ CMake options

  IF(CPUINFO_BUILD_UNIT_TESTS OR CPUINFO_BUILD_MOCK_TESTS)

    ENABLE_TESTING()

  

  error: patch failed: CMakeLists.txt:17

  error: CMakeLists.txt: patch does not apply

  Checking patch cmake/Config.cmake.in...

  Checking patch deps/clog/CMakeLists.txt...

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:144 (z_vcpkg_apply_patches)
  scripts/cmake/vcpkg_from_github.cmake:154 (vcpkg_extract_source_archive_ex)
  ports/cpuinfo/portfile.cmake:11 (vcpkg_from_github)
  scripts/ports.cmake:142 (include)

```

Picked those differences and created a new patch file.
What a sad that PR is not merged yet ... 

### Which triplets are supported/not supported? Have you updated the CI baseline?

No change in triplet support. The current build check must be passed. 

#### Target namespace changed

```
The package cpuinfo:... provides CMake targets:

    find_package(unofficial-cpuinfo CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::cpuinfo::clog unofficial::cpuinfo::cpuinfo)

```

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

I think so.

### CODE REVIEW

File: ports/cpuinfo/fix-install.patch [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
++++ b/CMakeLists.txt
+@@ -15,6 +15,28 @@ OPTION(CPUINFO_BUILD_UNIT_TESTS "Build cpuinfo unit tests" ON)
+ OPTION(CPUINFO_BUILD_MOCK_TESTS "Build cpuinfo mock tests" ON)
+ OPTION(CPUINFO_BUILD_BENCHMARKS "Build cpuinfo micro-benchmarks" ON)
+ 
++# Configuration
++set(config_install_dir "lib/cmake/${PROJECT_NAME}")
++set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
++set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
++set(targets_export_name "${PROJECT_NAME}Targets")
```
Comment (JackBoosY) [inline ports/cpuinfo/fix-install.patch:13]:
```suggestion
+set(targets_export_name "unofficial-${PROJECT_NAME}Targets")
```

File: ports/cpuinfo/fix-install.patch [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -15,6 +15,28 @@ OPTION(CPUINFO_BUILD_UNIT_TESTS "Build cpuinfo unit tests" ON)
+ OPTION(CPUINFO_BUILD_MOCK_TESTS "Build cpuinfo mock tests" ON)
+ OPTION(CPUINFO_BUILD_BENCHMARKS "Build cpuinfo micro-benchmarks" ON)
+ 
++# Configuration
++set(config_install_dir "lib/cmake/${PROJECT_NAME}")
++set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
++set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
```
Comment (JackBoosY) [inline ports/cpuinfo/fix-install.patch:12]:
```suggestion
+set(project_config "${generated_dir}/unofficial-${PROJECT_NAME}Config.cmake")
```

File: ports/cpuinfo/portfile.cmake [hunk 3]
```diff
@@ -1,45 +1,37 @@
...
         -DCPUINFO_BUILD_UNIT_TESTS=OFF
         -DCPUINFO_BUILD_MOCK_TESTS=OFF
         -DCPUINFO_BUILD_BENCHMARKS=OFF
 )
-
 vcpkg_install_cmake()
-
 vcpkg_copy_pdbs()
-
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
```
Comment (JackBoosY) [inline ports/cpuinfo/portfile.cmake:35]:
```suggestion
vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/unofficial-${PORT} TARGET_PATH unofficial-${PORT})
```

File: ports/cpuinfo/fix-install.patch [hunk 4]
```diff
@@ -0,0 +1,81 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e2d7d53..6d380e9 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -15,6 +15,28 @@ OPTION(CPUINFO_BUILD_UNIT_TESTS "Build cpuinfo unit tests" ON)
+ OPTION(CPUINFO_BUILD_MOCK_TESTS "Build cpuinfo mock tests" ON)
+ OPTION(CPUINFO_BUILD_BENCHMARKS "Build cpuinfo micro-benchmarks" ON)
+ 
++# Configuration
++set(config_install_dir "lib/cmake/${PROJECT_NAME}")
```
Comment (JackBoosY) [inline ports/cpuinfo/fix-install.patch:10]:
```suggestion
+set(config_install_dir "lib/cmake/unofficial-${PROJECT_NAME}")
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Since this is no the upstream changes, we should add prefix `unofficial-`.

**Comment by luncliff** [comment]
Cool. There are some ports I worked to use this, but let me fix them after this

**Comment by JackBoosY** [comment]
I think you should remake the patch file.

**Comment by luncliff** [comment]
Merged `master` to resolve failures of #16346 #16344

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
I think that that's incorrect; they're just corrections of upstream targets, so they should be under the upstream name, not `unofficial-`; @JackBoosY do you disagree?

**Comment by JackBoosY** [comment]
@strega-nil-ms The upstream doesn't export any targets in the master branch, did I miss something?

**Comment by strega-nil-ms** [comment]
@JackBoosY oh... yeah you're right, I totally misread that, you're right.

**Comment by strega-nil-ms** [comment]
Thanks @luncliff :)

**Pull request merged**